test/tests/functional/pbs_eligible_time.py

```diff
@@ -0,0 +1,103 @@
+# A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# The PBS Pro software is licensed under the terms of the GNU Affero General
+# Public License agreement ("AGPL"), except where a separate commercial license
+# agreement for PBS Pro version 14 or later has been executed in writing with
+# Altair.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software - under
+# a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestEligibleTime(TestFunctional):
+    """
+    Test suite for eligible time tests
+    """
+
+    def test_qsub_a(self):
+        """
+        Test that jobs requsting qsub -a <time> do not accrue
+        eligible time until <time> is reached
+        """
+        a = {'eligible_time_enable': 'True', 'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+
+        # Calculate time which qsub -a expects.  The strftime()
+        # will truncate the seconds, so the job will likely start in
+        # 1 min instead of 2.
+        now = int(time.time())
+        now += 120
+        s = time.strftime("%H%M", time.localtime(now))
+
+        J1 = Job(TEST_USER, attrs={ATTR_a: s})
+        jid = self.server.submit(J1)
+        self.server.expect(JOB, {ATTR_state: 'W'}, id=jid)
+
+        self.logger.info("Sleeping until job is out of 'W' state")
+        time.sleep(120)
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=jid)
+        self.server.expect(JOB, {'eligible_time': 60}, op=LT)
+
+    def test_job_array(self):
+        """
+        Test that a job array switches from accruing eligible time
+        to ineligible time when its last subjob starts running
+        """
+        a = {'eligible_time_enable': 'True', 'log_events': 2047}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        a = {'resources_available.ncpus': 2}
+        self.server.manager(MGR_CMD_SET, NODE, a, id=self.mom.shortname)
+
+        J1 = Job(TEST_USER, attrs={ATTR_J: '1-3'})
+        J1.set_sleep_time(20)
+        jid = self.server.submit(J1)
+        jid_short = jid.split('[')[0]
+        sjid1 = jid_short + '[1]'
+        sjid2 = jid_short + '[2]'
+        sjid3 = jid_short + '[3]'
+
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid1, extend='t')
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=sjid2, extend='t')
+        self.server.expect(JOB, {ATTR_state: 'Q'}, id=sjid3, extend='t')
```

Comment:
why not using 'job_state' instead of ATTR_state?
Also extend the test to also check that accrue_type is 2 (eligible_time)

Reply:
ATTR_state is the PTL internal constant for 'job_state'.  The idea is if in the future we change the attribute name from job_state to something else, the test doesn't need to change.